### PR TITLE
Feature/bp 255/integrating the spawner

### PR DESCRIPTION
**Pull request by engdorm**
new feature to run nodes inside containers
with zmq port to communicate with the different nodes and the movai-cli

### CODE REVIEW

**Comment by Mograbi** [comment]
@engdorm shouldn't this be opened for dev first ?

**Comment by AlexFernandes-MOVAI** [comment]
I suggest to make a python library out of container_tools module as it is shared in between movai-service and flow-initiator

File: flow_initiator/spawner/container_tools/orchestrator.py [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+            network_id: network id for the container, instead of network id
+            command: optional command to run the container
+            ports: open network ports for this container, list of (port: [ip, port])
+            auto_remove (bool): if to remove the container, when it's stopped.
+            labels (dict): label for the container
+            mounts: mounts drive option
+            env (list): environment variables for the container
+            volumes (dict): volumes to load
+            privileged (bool): if to run container in privileged mode
+            devices (list): devices to give permission
```
Comment (AlexFernandes-MOVAI) [inline flow_initiator/spawner/container_tools/orchestrator.py:73]: privileged, cap_add, devices ... how do we actually set hardware access ?

we can get inspiration by this kind of workflow : https://www.balena.io/docs/reference/base-images/base-images/#how-the-images-work-at-runtime 

check https://www.balena.io/

**Review by AlexFernandes-MOVAI — COMMENTED** [review]
Nice to finally see progress on this feature, I just commented first on the parts I am more concerned about.

Main remarks
- Please provide some indication on how to test the PR
- Please define what a flow is and how it is configured now (sample file)
- How containers access to hardware are defined ( privileged mode, cap_add, devices ... ) ?
- How do spawner/elements interact with hardware update : if a device changes name / disappear ? does the entrypoint of container need special care ?
- Will there be a CLI and/or API to control the flow-initiator ?
- What about ACL ?

File: flow_initiator/spawner/container_tools/__init__.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from .ttdocker_client import TTDockerClient
```
[the remark this answers is not in the document]
Comment (engdorm) [reply flow_initiator/spawner/container_tools/__init__.py:1]: this is why I have the orchestrator 
if the orchestrator will load different ContainerManager class (with different tech) 
the user of the orchestrator will not know this

**Pull request closed without merging**